Windows 7 support

rihardsgravis opened the pull request:
# Description

Mock Realm storage and revert to local storage

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

Tested manually on Window 7

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

Commit:
Mock Realm storage

Commit:
Add state storage encryption

Commit:
Set migration to complete by default

Commit:
Remove unused var

Commit:
Fix shared tests

rajivshah3 commented on src/desktop/native/preload/Electron.js:
```diff
@@ -194,15 +197,30 @@ const Electron = {
+    },
+
+    /**
+     * Remove all epracated redux-persist local storage items
```
```suggestion
     * Remove all deprecated redux-persist local storage items
```

rajivshah3 commented on src/desktop/native/preload/Electron.js:
```diff
@@ -194,15 +197,30 @@ const Electron = {
 
     /**
-     * Get all local storage items
+     * Get all depracated redux-persist local storage items
```
```suggestion
     * Get all deprecated redux-persist local storage items
```

rajivshah3 commented on src/desktop/native/Realm.js:
```diff
@@ -1,7 +1,14 @@
-// @see https://github.com/realm/realm-js/blob/master/lib/submit-analytics.js#L28
-process.env.REALM_DISABLE_ANALYTICS = 'true';
+/**
+ * Realm database mock for Windows 7 compataibillity
```
```suggestion
 * Realm database mock for Windows 7 compatibility
```

rajivshah3 committed:
Update src/desktop/native/preload/Electron.js

Co-Authored-By: rihardsgravis <31288628+rihardsgravis@users.noreply.github.com>

rajivshah3 committed:
Update src/desktop/native/Realm.js

Co-Authored-By: rihardsgravis <31288628+rihardsgravis@users.noreply.github.com>

rajivshah3 committed:
Update src/desktop/native/preload/Electron.js

Co-Authored-By: rihardsgravis <31288628+rihardsgravis@users.noreply.github.com>

cvarley100 committed:
Desktop: Update Windows 7 branch to latest (#1352)

* Bump electron from 3.1.6 to 3.1.8 in /src/desktop (#1294)

Bumps [electron](https://github.com/electron/electron) from 3.1.6 to 3.1.8.
- [Release notes](https://github.com/electron/electron/releases)
- [Commits](https://github.com/electron/electron/compare/v3.1.6...v3.1.8)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* feat: make sure we do not reconstruct and validate stored reattachments (#1293)

* Mobile: Fix seed text input issues (#1291)

* Mobile: Remove push to mask and fix Android letter duplication

* Mobile: Remove unused textinput prop

* Auto-retry with increased timeout for attachToTangle & getTransactionsToApprove APIs (#1285)

* Mobile/iOS: Rebuild Entangled framework (#1280)

Ref: rajivshah3/entangled@b246438

* Mobile/Android: Rebuild Entangled libs (#1281)

Ref: iotaledger/entangled@771bb1e

* Bump @babel/plugin-proposal-object-rest-spread in /src/desktop (#1305)

Bumps [@babel/plugin-proposal-object-rest-spread](https://github.com/babel/babel) from 7.4.0 to 7.4.3.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/master/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/compare/v7.4.0...v7.4.3)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump webpack-cli from 3.2.3 to 3.3.0 in /src/desktop (#1304)

Bumps [webpack-cli](https://github.com/webpack/webpack-cli) from 3.2.3 to 3.3.0.
- [Release notes](https://github.com/webpack/webpack-cli/releases)
- [Changelog](https://github.com/webpack/webpack-cli/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-cli/compare/V.3.2.3...v.3.3.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump mini-css-extract-plugin from 0.5.0 to 0.6.0 in /src/desktop (#1295)

Bumps [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin) from 0.5.0 to 0.6.0.
- [Release notes](https://github.com/webpack-contrib/mini-css-extract-plugin/releases)
- [Changelog](https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/mini-css-extract-plugin/compare/v0.5.0...v0.6.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump @babel/register from 7.0.0 to 7.4.0 in /src/desktop (#1297)

Bumps [@babel/register](https://github.com/babel/babel) from 7.0.0 to 7.4.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/master/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/compare/v7.0.0...v7.4.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Mobile: Use native dropdown component (#1275)

* Shared: Remove 2FA from all versions (#1292)

* Mobile: Remove 2FA

* Shared: Remove 2FA strings and actions/reducers

* Desktop: Remove 2fa

* Desktop: Revert shrinkwrap changes

* Desktop: Add initVault method

* Update lockfiles

* New Crowdin translations (#1277)

* New translations translation.json (Polish)

* New translations translation.json (Polish)

* New translations translation.json (Croatian)

* New translations translation.json (German)

* New translations translation.json (Croatian)

* New translations translation.json (Japanese)

* New translations translation.json (Japanese)

* New translations translation.json (Swedish)

* New translations translation.json (Swedish)

* Mobile: Fix navigation crash (#1308)

* Shared: Rebuild icon font file (#1309)

* Shared: Rebuild icon file

* Mobile: Replace Android icon file

* New Crowdin translations (#1307)

* New translations translation.json (Bulgarian)

* New translations translation.json (Spanish, Latin America)

* New translations translation.json (Vietnamese)

* New translations translation.json (Urdu (Pakistan))

* New translations translation.json (Thai)

* New translations translation.json (Turkish)

* New translations translation.json (Tamil)

* New translations translation.json (Finnish)

* New translations translation.json (Estonian)

* New translations translation.json (Dutch)

* New translations translation.json (Danish)

* New translations translation.json (Czech)

* New translations translation.json (Croatian)

* New translations translation.json (Chinese Traditional)

* New translations translation.json (Arabic)

* New translations translation.json (Swedish)

* New translations translation.json (Georgian)

* New translations translation.json (Polish)

* New translations translation.json (Korean)

* New translations translation.json (German)

* New translations translation.json (Chinese Simplified)

* New translations translation.json (French)

* New translations translation.json (German, Switzerland)

* New translations translation.json (Spanish)

* New translations translation.json (Norwegian)

* New translations translation.json (Slovenian)

* New translations translation.json (Slovak)

* New translations translation.json (Serbian (Latin))

* New translations translation.json (Russian)

* New translations translation.json (Romanian)

* New translations translation.json (Portuguese, Brazilian)

* New translations translation.json (Portuguese)

* New translations translation.json (Persian)

* New translations translation.json (Lithuanian)

* New translations translation.json (Greek)

* New translations translation.json (Latvian)

* New translations translation.json (Kannada)

* New translations translation.json (Japanese)

* New translations translation.json (Italian)

* New translations translation.json (Indonesian)

* New translations translation.json (Hungarian)

* New translations translation.json (Hindi)

* New translations translation.json (Hebrew)

* chore: disable redux-logger by default (#1310)

* Mobile: Fix SeedVault export name (#1312)

* Mobile: Fix unit info modal theming (#1311)

* Bump snyk from 1.150.0 to 1.151.0 (#1320)

* Bump i18next from 15.0.4 to 15.0.9 in /src/desktop (#1321)

* Bump prettier from 1.11.0 to 1.16.4 (#1248)

* Bump eslint-plugin-import from 2.16.0 to 2.17.1 (#1318)

* Desktop: Show modal for Ledger udev errors (#1192)

* Desktop: Show modal for Ledger udev errors

* Desktop: Localize udev error strings

* - Switch to link element for Ledger support url
- Add back to Seed intro button to udeverror popup

* Bump keytar from 4.4.0 to 4.4.1 in /src/desktop (#1316)

Bumps [keytar](https://github.com/atom/node-keytar) from 4.4.0 to 4.4.1.
- [Release notes](https://github.com/atom/node-keytar/releases)
- [Commits](https://github.com/atom/node-keytar/compare/v4.4.0...v4.4.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump webpack from 4.29.6 to 4.30.0 in /src/desktop (#1317)

Bumps [webpack](https://github.com/webpack/webpack) from 4.29.6 to 4.30.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v4.29.6...v4.30.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump react-router-dom from 4.3.1 to 5.0.0 in /src/desktop (#1319)

Bumps [react-router-dom](https://github.com/ReactTraining/react-router) from 4.3.1 to 5.0.0.
- [Release notes](https://github.com/ReactTraining/react-router/releases)
- [Commits](https://github.com/ReactTraining/react-router/compare/v4.3.1...v5.0.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Fix Linux icon path (#1322)

* Shared: Fix realm schema migration (#1326)

* New Crowdin translations (#1314)

* New translations translation.json (German)

* New translations translation.json (German)

* New translations translation.json (German)

* New translations translation.json (Portuguese, Brazilian)

* New translations translation.json (Portuguese, Brazilian)

* New translations translation.json (Polish)

* New translations translation.json (Croatian)

* New translations translation.json (Bulgarian)

* New translations translation.json (Spanish, Latin America)

* New translations translation.json (Vietnamese)

* New translations translation.json (Urdu (Pakistan))

* New translations translation.json (Thai)

* New translations translation.json (Turkish)

* New translations translation.json (Tamil)

* New translations translation.json (Finnish)

* New translations translation.json (Estonian)

* New translations translation.json (Dutch)

* New translations translation.json (Danish)

* New translations translation.json (Czech)

* New translations translation.json (Croatian)

* New translations translation.json (Chinese Traditional)

* New translations translation.json (Arabic)

* New translations translation.json (Swedish)

* New translations translation.json (Georgian)

* New translations translation.json (Polish)

* New translations translation.json (Korean)

* New translations translation.json (German)

* New translations translation.json (Chinese Simplified)

* New translations translation.json (French)

* New translations translation.json (German, Switzerland)

* New translations translation.json (Spanish)

* New translations translation.json (Norwegian)

* New translations translation.json (Slovenian)

* New translations translation.json (Slovak)

* New translations translation.json (Serbian (Latin))

* New translations translation.json (Russian)

* New translations translation.json (Romanian)

* New translations translation.json (Portuguese, Brazilian)

* New translations translation.json (Portuguese)

* New translations translation.json (Persian)

* New translations translation.json (Lithuanian)

* New translations translation.json (Greek)

* New translations translation.json (Latvian)

* New translations translation.json (Kannada)

* New translations translation.json (Japanese)

* New translations translation.json (Italian)

* New translations translation.json (Indonesian)

* New translations translation.json (Hungarian)

* New translations translation.json (Hindi)

* New translations translation.json (Hebrew)

* New translations translation.json (Italian)

* New translations translation.json (Japanese)

* New translations translation.json (Polish)

* Mobile: Upgrade react-native-keychain to 3.0.0 (#1180)

* Mobile: Upgrade react-native-keychain to 3.0.0

* Mobile/iOS: Release foundTypeRefs when they are no longer being used

Applies patch from oblador/react-native-keychain#188

* Mobile: Fix password fields crash (#1328)

* Mobile: Bump build no to 54 (#1325)

* Bump react-dom from 16.8.2 to 16.8.6 in /src/desktop (#1335)

Bumps [react-dom](https://github.com/facebook/react/tree/HEAD/packages/react-dom) from 16.8.2 to 16.8.6.
- [Release notes](https://github.com/facebook/react/releases)
- [Changelog](https://github.com/facebook/react/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/react/commits/v16.8.6/packages/react-dom)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump @babel/plugin-proposal-class-properties in /src/desktop (#1332)

Bumps [@babel/plugin-proposal-class-properties](https://github.com/babel/babel) from 7.3.3 to 7.4.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/master/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/compare/v7.3.3...v7.4.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump dart-sass from 1.17.2 to 1.18.0 in /src/desktop (#1331)

Bumps [dart-sass](https://github.com/sass/dart-sass) from 1.17.2 to 1.18.0.
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.17.2...1.18.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump kdbxweb from 1.2.6 to 1.2.7 in /src/desktop (#1333)

Bumps [kdbxweb](https://github.com/keeweb/kdbxweb) from 1.2.6 to 1.2.7.
- [Release notes](https://github.com/keeweb/kdbxweb/releases)
- [Changelog](https://github.com/keeweb/kdbxweb/blob/master/release-notes.md)
- [Commits](https://github.com/keeweb/kdbxweb/compare/1.2.6...1.2.7)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump eslint-plugin-import from 2.17.1 to 2.17.2 (#1337)

* Desktop: Trigger notifications for new transactions only (#1323)

* Desktop: Check for valid SeedVault file format (#1324)

* Check for valid SeedVault file format on import

* Fix string name formatting

* Bump snyk from 1.151.0 to 1.151.1 (#1345)

* Desktop: Update shrinkwrap (#1343)

* Bump react-markdown from 4.0.6 to 4.0.8 in /src/desktop (#1347)

Bumps [react-markdown](https://github.com/rexxars/react-markdown) from 4.0.6 to 4.0.8.
- [Release notes](https://github.com/rexxars/react-markdown/releases)
- [Changelog](https://github.com/rexxars/react-markdown/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rexxars/react-markdown/commits/v4.0.8)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump react from 16.8.3 to 16.8.6 in /src/desktop (#1346)

Bumps [react](https://github.com/facebook/react/tree/HEAD/packages/react) from 16.8.3 to 16.8.6.
- [Release notes](https://github.com/facebook/react/releases)
- [Changelog](https://github.com/facebook/react/blob/master/CHANGELOG.md)
- [Commits](https://github.com/facebook/react/commits/v16.8.6/packages/react)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Bump react-transition-group from 2.9.0 to 4.0.0 in /src/desktop (#1349)

Bumps [react-transition-group](https://github.com/reactjs/react-transition-group) from 2.9.0 to 4.0.0.
- [Release notes](https://github.com/reactjs/react-transition-group/releases)
- [Changelog](https://github.com/reactjs/react-transition-group/blob/master/CHANGELOG.md)
- [Commits](https://github.com/reactjs/react-transition-group/compare/v2.9.0...v4.0.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Desktop: Update README.md (#1338)

* Update README.md

* Update README.md

Commit:
Merge branch 'develop' into window-7

* develop: (127 commits)
  Mobile: Fix deep link behaviour (#1444)
  Bump snyk from 1.161.0 to 1.161.1 (#1455)
  Bump nyc from 13.1.0 to 14.1.0 in /src/shared (#1451)
  Bump snyk from 1.158.0 to 1.161.0 (#1448)
  Bump husky from 2.1.0 to 2.2.0 (#1447)
  Fix Electron API usage for Windows deep link functionality (#1438)
  Mobile: Bump build no to 61 (#1443)
  Mobile: Fix crash on empty password fields (#1441)
  Mobile: Bump build to 60 (#1437)
  New Crowdin translations (#1428)
  Convert all nested properties of address data from RealmObject to plain object (#1431)
  Shared: Add SeedVault export validation check (#1432)
  Mobile: Fix Android splash screen hang when returning from bg (#1429)
  Shared: Fix milestone fall behind check (#1436)
  Bump @babel/plugin-proposal-class-properties in /src/desktop (#1425)
  Bump @babel/preset-env from 7.4.3 to 7.4.4 in /src/desktop (#1415)
  [Security] Bump tar from 2.2.1 to 4.4.8 in /src/desktop (#1434)
  Bump snyk from 1.156.0 to 1.158.0 (#1435)
  Bump snyk from 1.154.0 to 1.156.0 (#1433)
  Bump @babel/register from 7.4.0 to 7.4.4 in /src/desktop (#1426)
  ...

# Conflicts:
#	package.json
#	src/desktop/main.js
#	src/desktop/npm-shrinkwrap.json
#	src/desktop/package.json
#	src/mobile/android/app/build.gradle
#	src/mobile/ios/iotaWallet-tvOS/Info.plist
#	src/mobile/ios/iotaWallet-tvOSTests/Info.plist
#	src/mobile/ios/iotaWallet.xcodeproj/project.pbxproj
#	src/mobile/ios/iotaWallet/Info.plist
#	src/mobile/ios/iotaWalletTests/Info.plist
#	src/mobile/ios/iotaWalletUITests/Info.plist
#	src/mobile/package.json
#	src/mobile/src/ui/components/PasswordFields.js
#	src/mobile/src/ui/views/wallet/Login.js
#	src/mobile/yarn.lock
#	src/shared/locales/ar/translation.json
#	src/shared/locales/bg/translation.json
#	src/shared/locales/cs/translation.json
#	src/shared/locales/da/translation.json
#	src/shared/locales/de-CH/translation.json
#	src/shared/locales/de/translation.json
#	src/shared/locales/el/translation.json
#	src/shared/locales/en/translation.json
#	src/shared/locales/es-ES/translation.json
#	src/shared/locales/es-LA/translation.json
#	src/shared/locales/et/translation.json
#	src/shared/locales/fa/translation.json
#	src/shared/locales/fi/translation.json
#	src/shared/locales/fr/translation.json
#	src/shared/locales/he/translation.json
#	src/shared/locales/hi/translation.json
#	src/shared/locales/hr/translation.json
#	src/shared/locales/hu/translation.json
#	src/shared/locales/id/translation.json
#	src/shared/locales/it/translation.json
#	src/shared/locales/ja/translation.json
#	src/shared/locales/ka/translation.json
#	src/shared/locales/kn/translation.json
#	src/shared/locales/ko/translation.json
#	src/shared/locales/lt/translation.json
#	src/shared/locales/lv/translation.json
#	src/shared/locales/nl/translation.json
#	src/shared/locales/no/translation.json
#	src/shared/locales/pl/translation.json
#	src/shared/locales/pt-BR/translation.json
#	src/shared/locales/pt-PT/translation.json
#	src/shared/locales/ro/translation.json
#	src/shared/locales/ru/translation.json
#	src/shared/locales/sk/translation.json
#	src/shared/locales/sl/translation.json
#	src/shared/locales/sr/translation.json
#	src/shared/locales/sv-SE/translation.json
#	src/shared/locales/ta/translation.json
#	src/shared/locales/th/translation.json
#	src/shared/locales/tr/translation.json
#	src/shared/locales/ur/translation.json
#	src/shared/locales/vi/translation.json
#	src/shared/locales/zh-CN/translation.json
#	src/shared/locales/zh-TW/translation.json
#	yarn.lock

Commit:
Version bump to 0.5.0-RC2

Commit:
Skip encryption if no persisted data present

Commit:
Update missed function rename

Commit:
Merge branch 'develop' into window-7

* develop:
  Desktop: Settings and Onboarding fixes (#1457)
  Mobile/Android: Rebuild Entangled libs (#1464)
  Bump dart-sass from 1.19.0 to 1.20.1 in /src/desktop (#1469)
  Bump webpack-cli from 3.3.1 to 3.3.2 in /src/desktop (#1472)
  Bump electron from 3.1.8 to 3.1.9 in /src/desktop (#1475)
  Shared: [Snyk] Fix for 1 vulnerable dependencies (#1474)
  Mobile: [Snyk] Fix for 1 vulnerable dependencies (#1473)
  Bump eslint-plugin-react-native from 3.2.1 to 3.7.0 (#1389)
  Bump fastlane from 2.121.1 to 2.122.0 in /src/mobile/android (#1468)
  Bump fastlane from 2.121.1 to 2.122.0 in /src/mobile/ios (#1467)
  Bump eslint-plugin-react from 7.12.4 to 7.13.0 (#1460)
  Bump lint-staged from 8.1.5 to 8.1.6 (#1461)

Commit:
Remove random selection of node on app start up

Commit:
Merge branch 'develop' into feat/nodes-granularity

Commit:
Make quorumSize configurable

quorumSize is currently used as a global constant throughout the quorum implementation. This is limiting especially for cases where we want quorumSize to be used as a user-defined variable.

This commit updates the quorum implementation and accepts quorumSize as a configurable variable.

rihardsgravis committed:
Raise an storage incompatibility error on Windows 7 (#1477)

Commit:
Merge branch 'develop' into window-7

* develop:
  Bump snyk from 1.161.1 to 1.162.0 in /src/shared (#1482)
  Bump snyk from 1.161.1 to 1.162.0 (#1481)
  Bump @babel/plugin-proposal-export-namespace-from in /src/shared (#1450)
  Desktop 0.5.0 Release (#1479)

# Conflicts:
#	src/desktop/package.json
#	src/desktop/src/ui/global/About.js

Commit:
Update auto update endpoint

Commit:
Merge branch 'develop' into feat/nodes-granularity

Commit:
Merge branch 'develop' into window-7

* develop:
  Bump snyk from 1.162.0 to 1.163.0 (#1491)
  New Crowdin translations (#1446)
  Desktop: Add migration explanation (#1445)
  Add issue template metadata [ci skip] (#1486)
  Update issue templates to newer version (#1484)
  Shared: Bump update versioning (#1478)

Commit:
Merge branch 'develop' into window-7

* develop:
  Shared: Fix extra parameter in fn call when reattaching and promoting (#1503)
  Mobile: Bump handlebars to 4.1.2 (#1454)
  Mobile/iOS: Patch react-native run-ios (#1505)
  Bump url-parse from 1.4.3 to 1.4.7 in /src/shared (#1453)
  Bump proxy-polyfill from 9c009d7 to 0.3.0 in /src/shared (#1449)
  Bump snyk from 1.163.1 to 1.163.3 (#1499)
  Bump react-transition-group from 4.0.0 to 4.0.1 in /src/desktop (#1497)
  - Update Bugsnag dependency usage (#1493)
  Bump snyk from 1.163.0 to 1.163.1 (#1495)

# Conflicts:
#	src/desktop/src/index.js
#	src/shared/store.js

cvarley100 committed:
Mobile: Add basic node settings UI shell

Commit:
Merge branch 'develop' into feat/nodes-granularity

rajivshah3 committed:
Merge branch 'develop' into feat/node-granularity-ui

Commit:
feat: add nodes manager for handling network requests to IRI nodes

Commit:
feat: add state variables & actions for quorum configuration

Commit:
feat: add state variables & actions for node auto-switching

Commit:
feat: update realm schema for authentication key

Commit:
feat: update realm schema for quorum configuration

rihardsgravis committed:
Desktop: Temporary pre-build Trinity on Windows 7 (#1510)

* Temporary pre-build Trinity javascript application

* Update Bugsnag API key after building the js app

rajivshah3 committed:
Desktop: Avoid writing to stdout when replacing Bugsnag API key

Commit:
Remove prebuild and update build script

Commit:
Temporary add `build-no-concurrent` build script

Commit:
feat: add state variables & actions for node auto nost list

cvarley100 committed:
Mobile: Add mock node settings and update custom node UI

Commit:
feat: do not retry on different nodes if auto node switching is disabled

Commit:
feat: add wallet storage methods for new node improvements

cvarley100 committed:
Mobile: Update toggle styling

cvarley100 committed:
Mobile: Fix addCustomNode layout animation

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

cvarley100 committed:
Merge branch 'feat/nodes-granularity' into feat/node-granularity-ui

cvarley100 committed:
Mobile: Update node settings

cvarley100 committed:
Mobile: Settings refactor

Commit:
Merge branch 'develop' into feat/nodes-granularity

Commit:
feat: minor updates

- Replace deprecated #withRetriesOnDifferentNodes with NodesManager.withRetries
- Change the way nodes data is stored in redux. Also update corresponding actions.
- Allow caller of nodesConfigurationFactory to override nodes

Commit:
Initial node granularity UI implementation

Commit:
Fix linting errors

cvarley100 committed:
Merge branch 'feat/nodes-granularity' into feat/node-granularity-ui

cvarley100 committed:
Mobile: Fix automatic node mangagement setting

Commit:
feat: make sure all utils from extendedApi module accept provider settings as first parameter

Commit:
feat: make sure all wallet actions are wrapped with NodeManager.withRetries

Commit:
feat: Cancel auto-retry on (some) application generated errors

Commit:
feat: Automatically retry on different nodes on failed transaction action

Commit:
feat: add password property in nodes data

Commit:
feat: make sure @iota/core methods are passed with username & password

Commit:
feat: make sure promotion & reattachment actions are wrapped inside NodeManager.withRetries method

Commit:
feat: make sure prepareTransfer method of SeedStore class accept node settings parameter

Commit:
Merge branch 'develop' into feat/nodes-granularity

Commit:
fix: update/fix failing tests

Commit:
Merge branch 'feat/nodes-granularity' into feat/node-granularity-desktop

* feat/nodes-granularity: (48 commits)
  fix: update/fix failing tests
  Bump timekeeper from 2.0.0 to 2.2.0 in /src/shared (#1507)
  Bump bignumber.js from 8.0.1 to 8.1.1 in /src/shared (#1545)
  New Crowdin translations (#1516)
  Desktop: Catch and display preload and initialisation errors (#1549)
  Remove auditjs (#1553)
  feat: make sure prepareTransfer method of SeedStore class accept node settings parameter
  Revert and lock to `usb` `1.5.0` (#1558)
  feat: make sure promotion & reattachment actions are wrapped inside NodeManager.withRetries method
  feat: make sure @iota/core methods are passed with username & password
  Desktop: Update ledger warning string (#1557)
  Bump @ledgerhq/hw-transport-node-hid in /src/desktop (#1555)
  Bump appcompat-v7 from 27.1.0 to 28.0.0 in /src/mobile/android (#961)
  Bump snyk from 1.165.0 to 1.165.1 (#1552)
  Desktop: Improve UX during Ledger setup (#1550)
  Shared: Improvements to alerts (#1488)
  feat: add password property in nodes data
  Shared: Add device time suggestion to out of sync errors (#1485)
  Bump eslint-config-prettier from 4.2.0 to 4.3.0 (#1546)
  Bump express from 4.16.4 to 4.17.0 in /src/desktop (#1547)
  ...

# Conflicts:
#	src/mobile/src/ui/views/wallet/ChangePassword.js

Commit:
Update node granularity UI

cvarley100 committed:
Resolve conflicts

Commit:
Merge branch 'feat/node-granularity-ui' into feat/node-granularity-desktop

* feat/node-granularity-ui:
  Mobile: Fix automatic node mangagement setting

# Conflicts:
#	src/mobile/src/ui/views/wallet/ChangePassword.js

Commit:
Linting fixes

cvarley100 committed:
Merge branch 'develop' into feat/node-granularity-ui

cvarley100 committed:
Mobile: Comment out redux logger

Commit:
fix: make sure iota.settings object has provider property

cvarley100 committed:
Merge branch 'feat/nodes-granularity' into feat/node-granularity-ui

cvarley100 committed:
Merge branch 'feat/node-granularity-desktop' into feat/node-granularity-ui

cvarley100 committed:
Shared: Fix various issues with node settings

cvarley100 committed:
Shared: Minor custom node fixes

cvarley100 committed:
Shared: Fix incorrect node config

Commit:
fix: do not mutate state.settings.quorum object

cvarley100 committed:
Merge branch 'feat/nodes-granularity' into feat/node-granularity-ui

Commit:
feat: use redux store directly in NodesManager class

cvarley100 committed:
Mobile: Fix node settings being updated incorrectly

cvarley100 committed:
Shared: Fix node change alerts

cvarley100 committed:
Merge branch 'feat/nodes-granularity' into feat/node-granularity-ui

Commit:
- Remove `autoNodeSelection` from props
- Clear custom node input after adding custom node
- Update custom node list to node object use

cvarley100 committed:
Merge branch 'develop' into feat/node-granularity-ui

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

cvarley100 committed:
Mobile: Remove log statements

Commit:
Add quorum size checks

Commit:
Mobile: Add max quorum size and change node if updated from props

Commit:
Add `MAXIMUM_QUORUM_SIZE`

Commit:
Change state node on prop change

cvarley100 committed:
Mobile: Node settings comments and cleanup

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

cvarley100 committed:
Mobile: Adjust node settings comments

cvarley100 committed:
Mobile: Fix quorum size options and available nodes

Commit:
Update Primary node to a Select

Commit:
Merge branch 'feat/node-granularity-ui' of github.com-rihardsgravis:iotaledger/trinity-wallet into feat/node-granularity-ui

* 'feat/node-granularity-ui' of github.com-rihardsgravis:iotaledger/trinity-wallet:
  Mobile: Node settings comments and cleanup
  Mobile: Add max quorum size and change node if updated from props

Commit:
Remove deprecated `autoNodeSwitching`

cvarley100 committed:
erge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

Commit:
Fix lint errors

cvarley100 committed:
Shared: Fix node lists and quorum size

cvarley100 committed:
Shared: Fix quorum config storage in realm

cvarley100 committed:
Shared: Fix setting of quorum size

Commit:
Merge branch 'develop' into window-7

* develop: (49 commits)
  Desktop: Use executable icon for application window
  Desktop: Lock node-usb to 1.5.0 (#1580)
  Bump snyk from 1.165.1 to 1.166.0 (#1570)
  Bump @ledgerhq/hw-transport-node-hid in /src/desktop (#1562)
  Bump i18next from 15.1.1 to 15.1.2 in /src/desktop (#1561)
  Bump webpack from 4.31.0 to 4.32.0 in /src/desktop (#1564)
  Bump recharts from 1.6.0 to 1.6.1 in /src/desktop (#1567)
  Bump timekeeper from 2.0.0 to 2.2.0 in /src/shared (#1507)
  Bump bignumber.js from 8.0.1 to 8.1.1 in /src/shared (#1545)
  New Crowdin translations (#1516)
  Desktop: Catch and display preload and initialisation errors (#1549)
  Remove auditjs (#1553)
  Revert and lock to `usb` `1.5.0` (#1558)
  Desktop: Update ledger warning string (#1557)
  Bump @ledgerhq/hw-transport-node-hid in /src/desktop (#1555)
  Bump appcompat-v7 from 27.1.0 to 28.0.0 in /src/mobile/android (#961)
  Bump snyk from 1.165.0 to 1.165.1 (#1552)
  Desktop: Improve UX during Ledger setup (#1550)
  Shared: Improvements to alerts (#1488)
  Shared: Add device time suggestion to out of sync errors (#1485)
  ...

# Conflicts:
#	src/desktop/src/index.js
#	src/desktop/src/ui/global/FatalError.js
#	src/shared/store.js

Commit:
Fix wallet reset

cvarley100 committed:
Mobile: Clean up use of omit

rihardsgravis committed:
Desktop: Fix Windows 7 release wallet reset functionality (#1584)

* Fix Windows 7 reset issue

* Linting fixes

cvarley100 committed:
Mobile: Fix node selection list

Commit:
Merge branch 'feat/node-granularity-ui' into fix/remove-autonodeswitching

* feat/node-granularity-ui:
  Mobile: Fix node selection list
  Mobile: Clean up use of omit
  Shared: Fix setting of quorum size
  Shared: Fix quorum config storage in realm
  Shared: Fix node lists and quorum size
  Mobile: Fix quorum size options and available nodes
  Mobile: Adjust node settings comments

# Conflicts:
#	src/desktop/src/ui/views/settings/Node.js
#	src/mobile/src/ui/routes/entry.js
#	src/shared/containers/settings/Node.js

cvarley100 committed:
Mobile: Adjust quorum size update behaviour

Commit:
Shared: Move all action types to a separate file (#1578)

* chore: move action types to a separate file

This commit moves all action types to a separate directory /types to avoid circular dependencies.

* fix: update failing tests

Commit:
Merge branch 'feat/node-granularity-ui' into fix/remove-autonodeswitching

* feat/node-granularity-ui:
  Shared: Move all action types to a separate file (#1578)

# Conflicts:
#	src/shared/__tests__/reducers/settings.spec.js
#	src/shared/actions/settings.js
#	src/shared/reducers/settings.js

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

Commit:
Merge branch 'develop' into window-7

* develop:
  Bump @ledgerhq/hw-transport-node-hid from 4.58.0 to 4.60.0 in /src/desktop (#1589)
  Bump snyk from 1.166.0 to 1.166.1 (#1587)
  Mobile: Bump build numbers for release 0.7.1 (62) (#1586)
  Mobile: Fix long account name styling on transfer confirmation modal (#1569)
  Mobile: Fix and refactor PasswordFields (#1554)
  New Crowdin translations (#1559)
  Check for correct Realm encryption key length (#1585)

# Conflicts:
#	src/desktop/src/libs/realm.js

Commit:
Merge branch 'develop' into feat/node-granularity-ui

rihardsgravis committed:
Desktop Windows 7 release 0.5.1 (#1591)

* Version bump and Changelog update

* Update build script command

* Disable code minification

* Update About.js

* Update About.js

Commit:
Node selection improvement fixes (#1597)

* Desktop: Lock node-usb to 1.5.0 (#1580)

* Desktop: Use executable icon for application window

BrowserWindow can default to executable icon if no icon is set, which works nicely on all platforms.

* Check for correct Realm encryption key length (#1585)

* New Crowdin translations (#1559)

* New translations translation.json (German)

* New translations translation.json (Swedish)

* New translations translation.json (Norwegian)

* Mobile: Fix and refactor PasswordFields (#1554)

* Mobile: Fix long account name styling on transfer confirmation modal (#1569)

* Mobile: Fix long account name styling on transfer confirmation modal

* Update src/mobile/src/ui/components/TransferConfirmationModal.js

* Mobile: Bump build numbers for release 0.7.1 (62) (#1586)

* Bump snyk from 1.166.0 to 1.166.1 (#1587)

* Bump @ledgerhq/hw-transport-node-hid from 4.58.0 to 4.60.0 in /src/desktop (#1589)

* Bump @ledgerhq/hw-transport-node-hid in /src/desktop

Bumps [@ledgerhq/hw-transport-node-hid](https://github.com/LedgerHQ/ledgerjs) from 4.58.0 to 4.60.0.
- [Release notes](https://github.com/LedgerHQ/ledgerjs/releases)
- [Commits](https://github.com/LedgerHQ/ledgerjs/compare/v4.58.0...v4.60.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

* Desktop: Lock usb to 1.5.0

* Revert to usb `1.5.0`

* fix realm schema version 3 migration issues

* fix: resolve lint errors in desktop/

* fix failing tests (mobile & desktop)

* chore: upgrade iota.lib.js to latest version

cvarley100 committed:
Shared: Add not enough synced  nodes for quorum alert and refactor al… (#1577)

* Shared: Add not enough synced  nodes for quorum alert and refactor alerts

* Shared: Rename alert

Commit:
Merge remote-tracking branch 'upstream/feat/node-granularity-ui' into feat/node-granularity-ui

cvarley100 committed:
Shared: Fix failing tests

Commit:
Merge remote-tracking branch 'upstream/feat/node-granularity-ui' into feat/node-granularity-ui

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Mobile: Fix auth fields crash

Commit:
Merge remote-tracking branch 'upstream/feat/node-granularity-ui' into feat/node-granularity-ui

Commit:
fix: fallback to default node object if node list is empty

Commit:
fix: update extendedApi tests

cvarley100 committed:
Shared: Fix v3 migration

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

cvarley100 committed:
Mobile: Bump build no to 63

rajivshah3 committed:
Mobile: Bump build number to 64

rajivshah3 committed:
Mobile: Bump build numbers for release v0.7.2 (65)

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Mobile: Fix blank screen when changing node from loading

cvarley100 committed:
Mobile: Fix settings dual footer activityy indicator placement

cvarley100 committed:
Mobile: Clean up add custom node fields

Commit:
Mobile: Allow exporting complete settings from state (#1637)

* feat: expose quorum nodes & size

* feat: update state props in state export

This commit allows exporting `state.settings` & exposed values on global instances

cvarley100 committed:
Merge branch 'develop' into feat/node-granularity-ui

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

cvarley100 committed:
Mobile: Bump build no to 66

cvarley100 committed:
Resolve conflicts

Commit:
Shared: Block autoretry if a tx has a fatal error on retry (#1644)

* feat: limit broadcast loop

* Desktop: Link up autoretry limiting

* Shared: Update autoretry attempts on manual retry and only update when tx error is fatal

* Shared: Remove retry count and block autoretry on one fatal tx error

* feat: assign "fatalErrorOnRetry" to transactions when account info is synced

Commit:
fix: correctly call promote & reattach actions during manual promotion (#1648)

Commit:
Merge branch 'develop' into feat/node-granularity-ui

* develop:
  Bump @ledgerhq/hw-transport-node-hid from 4.60.0 to 4.60.2 in /src/desktop (#1612)
  Desktop test coverage (#1560)
  Bump enzyme-adapter-react-16 from 1.7.0 to 1.13.2 in /src/mobile (#1632)
  Bump snyk from 1.168.0 to 1.169.2 (#1650)

# Conflicts:
#	src/desktop/src/index.js
#	src/desktop/src/ui/Index.js

cvarley100 committed:
Shared: Downgrade @iota/core to 1.0.0-beta.11 (#1649)

* Shared: Revert iota core version

* Mobile: Update yarn.lock

cvarley100 committed:
Mobile: Temporarily disable basic auth (#1651)

cvarley100 committed:
Mobile: Bump build no to 67

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

Commit:
Update desktop tests

Commit:
Merge branch 'feat/node-granularity-ui' of github.com:iotaledger/trinity-wallet into feat/node-granularity-ui

* 'feat/node-granularity-ui' of github.com-rihardsgravis:iotaledger/trinity-wallet:
  Mobile: Bump build no to 67
  Mobile: Temporarily disable basic auth (#1651)

cvarley100 committed:
Desktop: Fix string

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

Commit:
version bump and Changelog update

Commit:
Lint fixes

Commit:
Revert bugsnag changes

Commit:
Add newline at end of file

Commit:
Fix `package.json` path

rihardsgravis committed:
Update src/desktop/src/ui/global/About.js

Co-Authored-By: Rajiv Shah <rajivshah1@icloud.com>

rihardsgravis committed:
Update src/desktop/src/ui/global/About.js

Co-Authored-By: Rajiv Shah <rajivshah1@icloud.com>

Commit:
- Disable devTools in production mode
- Set `/index.html` as default route

Commit:
Merge branch 'desktop-0.5.2' into window-7

* desktop-0.5.2: (44 commits)
  - Disable devTools in production mode - Set `/index.html` as default route
  Update src/desktop/src/ui/global/About.js
  Update src/desktop/src/ui/global/About.js
  Fix `package.json` path
  Add newline at end of file
  Revert bugsnag changes
  Lint fixes
  version bump and Changelog update
  Shared: Downgrade @iota/core to 1.0.0-beta.11 (#1649) (#1654)
  Desktop: Temporarily show error for Windows users with special characters (#1653)
  Bump @ledgerhq/hw-transport-node-hid from 4.60.0 to 4.60.2 in /src/desktop (#1612)
  Desktop test coverage (#1560)
  Bump enzyme-adapter-react-16 from 1.7.0 to 1.13.2 in /src/mobile (#1632)
  Bump snyk from 1.168.0 to 1.169.2 (#1650)
  fix: remove dead nodes (#1646)
  Add Snapshot Transition warning for existing seeds (#1643)
  Desktop: FatalError component Linux fix (#1638)
  Bump snyk from 1.167.2 to 1.168.0 (#1642)
  Bump snyk from 1.161.1 to 1.167.2 in /src/mobile (#1631)
  Bump jest from 24.0.0 to 24.8.0 in /src/mobile (#1614)
  ...

# Conflicts:
#	src/desktop/.gitignore
#	src/desktop/bugsnag.sh
#	src/desktop/package.json
#	src/desktop/src/index.js
#	src/desktop/src/ui/global/About.js

cvarley100 committed:
Merge branch 'develop' into feat/node-granularity-ui

Commit:
Temporary disable custom node authorisation entry

Commit:
Merge branch 'develop' into feat/node-granularity-ui

* develop:
  Deskop 0.5.2 release (#1656)
  Bump react-transition-group from 4.0.1 to 4.1.0 in /src/desktop (#1663)
  Desktop `0.5.2` version bump (#1668)
  Bump snyk from 1.169.2 to 1.170.0 (#1659)
  New Crowdin translations (#1593)
  Shared: Copy missing strings (#1655)
  Shared: Downgrade @iota/core to 1.0.0-beta.11 (#1649) (#1654)
  Desktop: Temporarily show error for Windows users with special characters (#1653)

Commit:
Update failing tests

Commit:
Bump version and update Changelog to `0.6.0-RC1`

Commit:
Desktop linting fixes

Commit:
Merge branch 'feat/node-granularity-ui' into desktop-0.6.0

* feat/node-granularity-ui:
  Desktop linting fixes

cvarley100 committed:
Mobile: Fix crashes/behaviour on node settings

cvarley100 committed:
Merge branch 'feat/node-granularity-ui' of https://github.com/iotaledger/trinity-wallet into feat/node-granularity-ui

cvarley100 committed:
Shared: Fix remote node list and out of sync alerts (#1672)

* Shared: Add missing strings

* Shared: Fix fetch remote node list functionality

* Shared: Adjust fetch remote node list behaviour

* Shared: Fix out of sync alerts

* Shared: Remove unnecessary dispatch

* Shared: Remove unnecessary dispatch

* Shared: Minor refactor

* Update src/shared/locales/en/translation.json

Commit:
Merge branch 'feat/node-granularity-ui' into desktop-0.6.0

* feat/node-granularity-ui:
  Shared: Fix remote node list and out of sync alerts (#1672)
  Mobile: Fix crashes/behaviour on node settings

Commit:
Merge branch 'desktop-0.6.0' into window-7

* desktop-0.6.0: (92 commits)
  Shared: Fix remote node list and out of sync alerts (#1672)
  Mobile: Fix crashes/behaviour on node settings
  Desktop linting fixes
  Bump version and update Changelog to `0.6.0-RC1`
  Update failing tests
  Deskop 0.5.2 release (#1656)
  Bump react-transition-group from 4.0.1 to 4.1.0 in /src/desktop (#1663)
  Temporary disable custom node authorisation entry
  Desktop `0.5.2` version bump (#1668)
  Bump snyk from 1.169.2 to 1.170.0 (#1659)
  New Crowdin translations (#1593)
  Shared: Copy missing strings (#1655)
  Desktop: Fix string
  Update desktop tests
  Mobile: Bump build no to 67
  Mobile: Temporarily disable basic auth (#1651)
  Shared: Downgrade @iota/core to 1.0.0-beta.11 (#1649)
  fix: correctly call promote & reattach actions during manual promotion (#1648)
  Shared: Block autoretry if a tx has a fatal error on retry (#1644)
  Mobile: Bump build no to 66
  ...

# Conflicts:
#	src/desktop/package.json
#	src/desktop/src/index.js
#	src/desktop/src/ui/global/About.js

rihardsgravis commented:
Closing this PR as it is not meant to be merged at any time.
